contracts/axelar-gateway/src/testutils.rs

```diff
@@ -174,3 +177,26 @@ pub fn generate_proof(env: &Env, data_hash: BytesN<32>, signer_set: TestSignerSe
         nonce: signer_set.signers.nonce,
     }
 }
+
+// auth utils
+
+pub fn generate_random_payload_and_hash(env: &Env) -> BytesN<32> {
+    let payload: Bytes = BytesN::<10>::random(env).into();
+    env.crypto().keccak256(&payload).into()
+}
+
+pub fn rotate_signers(env: &Env, contract_id: &Address, new_signers: TestSignerSet) {
+    let mut epoch_val: u64 = 0;
+    env.as_contract(&contract_id, || {
+        epoch_val = epoch(&env) + 1;
+        auth::rotate_signers(env, &new_signers.signers, false);
+    });
+
+    assert_emitted_event(
```

Comment:
please use `assert_last_emitted_event` instead of `assert_emitted_event` with -1 index.